### DISCUSSION TITLE
Confusion about speed up for sdxl+controlnet.

### DISCUSSION THREAD

**Discussion by weisy11** [lllyasviel/stable-diffusion-webui-forge#479] [Q&A]
Very good job! I'm planning to move my projects to using forge, but I'm not sure if it's for me. Has anybody tried to deploy forge on 4090 or A10? How is the speed up for SDXL + ControlNet? As I'm using the card with large vram, I want to figure out the speed up I can get. 
The [Readme](https://github.com/lllyasviel/stable-diffusion-webui-forge?tab=readme-ov-file#stable-diffusion-webui-forge) says the speed up is different on different gpu card. And it says speed with SDXL+ControlNet will speed up about 30\~45% without mentioning the gpu card. So does 30\~45% here means forge will speed up no matter what GPU card I use, or on the specified card, e.g. common GPU like 8GB vram. Will the speed up different if I use different combination of ControlNet? What is the common combination that will get 30~45% speed up?

**Comment by yamfun** [comment]
Forge reduced VRAM usage by say, X gb.
So if some people with some workflow that takes say, X gb VRAM more than the VRAM of their GPU and so triggered the "fallback to slow system ram" mechanism, then using Forge will make them avoid that and so they feel very fast improvement.
So someone with 4090 but a 17gb workflow vs someone with 4090 but a 25gb workflow, will feel very different improvement....

**Reply by weisy11** [reply]
That's right. So I want to figure out what is the **common** case to get the speed up. Maybe forge can provide more detailed data.

**Reply by yamfun** [reply]
What I mean is, so, you have to try it yourself because everyone is different. 

Forge is almost like A1111 so it is easy to try

**Reply by rotatorotator** [reply]
I have a 4090 as well. Forge needs less Vram, so you can go for higher resolution before running out of memory. Speed is slightly faster, probably 3-5% but i did not perform any statistical analysis.

**Comment by leomaxwell973** [comment]
All i gotta say is, I never thought I'd get AniDiff to actually pass 64 frames in one batch (12GB VRAM) back using A1111 vanilla. I actually had to sit down and just kinda stare, and marvel at the no OOM

come to think of it, forge seems to have the ethic of, OOMs are not by design, usually means somethings leaking. (which sadly, happens often, but hey, we all leak in our early days, amirite? :P)
Where-as vanilla A1111 convinces you OOM is life (not love)

afterthought: I actually deleted my vanilla for fresh install on forge... I might be wrong... but feels like it was really designed for the place on top the original approach, which, takes away half the risk, being required to migrate (unless you got bigbadaboom space for dayz), as inno uninstalls are required. But yea, there were growing pains doing it as the base, with things glitching, wonder if its better with the in-place install 🤔
